staging/src/k8s.io/apiserver/pkg/util/webhook/client_test.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func TestWebhookClientConfig(t *testing.T) {
+	cm, _ := NewClientManager([]schema.GroupVersion{})
+	authInfoResolver, err := NewDefaultAuthenticationInfoResolver("")
+	if err != nil {
+		t.Fatal(err)
+	}
+	cm.SetAuthenticationInfoResolver(authInfoResolver)
+	cm.SetServiceResolver(NewDefaultServiceResolver())
+
+	tests := []struct {
+		name             string
+		url              string
+		expectAllowHTTP2 bool
+	}{
+		{
+			name:             "force http1",
+			url:              "https://webhook.example.com",
+			expectAllowHTTP2: false,
+		},
+		{
+			name:             "allow http2 for localhost",
+			url:              "https://localhost",
+			expectAllowHTTP2: true,
+		},
+		{
+			name:             "allow http2 for 127.0.0.1",
+			url:              "https://127.0.0.1",
+			expectAllowHTTP2: true,
+		},
+		{
+			name:             "allow http2 for [::1]:0",
+			url:              "https://[::1]",
+			expectAllowHTTP2: true,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+
+			cc := ClientConfig{
+				URL: tc.url,
+			}
+			cfg, err := cm.hookClientConfig(cc)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if tc.expectAllowHTTP2 && len(cfg.NextProtos) != 0 {
```

Comment:
nit
```suggestion
			if tc.expectAllowHTTP2 && len(cfg.NextProtos) == 1 && cfg.NextProtos[0] == "http/1.1"{
```

NextProtos can be

[]string{"http/1.1"}
[]string{"h2"}
[]string{"h2", "http/1.1"}